### ISSUE TITLE
Latest update flagged as malware by macOS - won't open

### ISSUE THREAD

**Issue by hampton8** [lihaoyun6/AirBattery#46]
Just updated to the latest version, and macOS is blocking opening AirBattery.app, flagging it as malware.

Same as [this](https://github.com/lihaoyun6/QuickRecorder/issues/77) open issue for QuickRecorder, so seems to be an issue with your developer cert.

<img width="367" alt="SCR-20240616-czac" src="https://github.com/lihaoyun6/AirBattery/assets/75634978/e7b6ab95-b102-4725-a48a-99b6a54d510d">

**Comment by lihaoyun6** [comment]
<img width="1102" alt="image" src="https://github.com/lihaoyun6/QuickRecorder/assets/16348097/add8daf7-be30-4cb2-a597-3f6b52157c0f">
<img width="486" alt="image" src="https://github.com/lihaoyun6/QuickRecorder/assets/16348097/fe697201-f424-4575-9f3a-2c8ff2bf1022">

I finally figured out what the fuck happened here.
When I was programming in macOS 14, Apple issued me a certificate, but when I tried to program on macOS 15 + Xcode 16 Beta, Apple issued me a new certificate again and revoked the certificate belonging to macOS 14.
But it didn't tell me that the certificate had been revoked!!! And my keychain showed that the certificate was always valid!!!
However macOS only recognizes the **"Latest"** certificate, which means that I actually have an invalid certificate, but I don't know it's invalid. Why didn't Apple tell me this???

**Comment by lihaoyun6** [comment]
I have re-signed AirBattery with the correct certificate and re-uploaded the v1.3.8 release. You can now try downloading and re-installing it.

**Comment by hampton8** [comment]
Yep, all good now. Thank you, both AirBattery and QuickRecorder are fantastic apps :)

**Comment by lihaoyun6** [comment]
It's great to hear that everything is working fine 😇
It took me all night to figure out what was going on. I love Apple as much as I hate it.

**Comment by lihaoyun6** [comment]
emmm... sorry, I shouldn't close this issue, I should let more people see it

**Comment by abdelkaderboui** [comment]
The issue still persists on macOS 14.5. I am seeing the same issue with QuickRecorder.
<img width="372" alt="Screenshot 2024-06-16 at 13 17 04" src="https://github.com/lihaoyun6/AirBattery/assets/172973604/c395e52c-45bc-4566-a5e7-6a9b438feca8">

**Comment by cgensheimer** [comment]
yep, blocked on macOS 15.0, I will note this was the update installed by the auto updater.

**Comment by itsyoshio** [comment]
I currently have that issue via the homebrew tap. (I deleted my cache)

**Comment by Proxymiity** [comment]
> I have re-signed AirBattery with the correct certificate and re-uploaded the v1.3.8 release. You can now try downloading and re-installing it.

Is it possible to push that update to the homebrew tap? Thanks in advance 🙂

**Comment by nabinkhadka** [comment]
Yes, homebrew still giving. I can confirm that I was able to use it by downloading the dmg file.
<img width="649" alt="image" src="https://github.com/lihaoyun6/AirBattery/assets/6050613/838c1f59-b157-4567-8d2d-261f6faf41ff">

**Comment by rafinesse** [comment]
Can't run on macOS 15 (Beta) unfortunately, downloaded newest version (4 days old), can't override malware protection either.
<img width="255" alt="Screenshot 2024-06-20 at 11 17 43" src="https://github.com/lihaoyun6/AirBattery/assets/71700237/0faf5b85-22fc-4dd2-a0a2-fbff9950b832">

**Comment by synapse73** [comment]
i deleted the app, still show the warning
i reinstalled the latest version, still show the warning
need help

**Comment by synapse73** [comment]
> i deleted the app, still show the warning i reinstalled the latest version, still show the warning need help

after delete airbattery from login item, finally no poping of the warning...

**Comment by leylaneymat** [comment]
Even after downloading the latest version (1.4.7), I still get such error. Will it be fixed soon? I really like your app and would love to use it, but do not wanna risk my privacy. Thank you.

<img width="264" alt="Screenshot 2024-07-23 at 9 29 41 AM" src="https://github.com/user-attachments/assets/0e0659bc-a939-42cf-a8d3-96f5baa888e9">

**Comment by iceonfirek** [comment]
> > i deleted the app, still show the warning i reinstalled the latest version, still show the warning need help
> 
> after delete airbattery from login item, finally no poping of the warning...

You saved my ass.... almost gonna crazy by popping of this warning

**Comment by Tanzim9999** [comment]
I found a workaround for this issue,
After Installing AirBattery and try to open it once,
If you get the error of malware.
You need to goto Privacy & Security, Swipe down till you see AirBattery- Open Anyway- Give permission to open anyway.
Done!
